File: FATT_backend/WebAPI/Models/Exercise.cs
namespace WebAPI.Models
{
    public class Exercise
    {
        public Exercise()
        {
            this.Workouts = new HashSet<Workout>();
        }
        public long Id { get; set; }
        public string Name { get; set; } = "";
        public string Category { get; set; } = "";
        public string Intensity { get; set; } = "";
        public string Equipment { get; set; } = "";
        public string Repetitions { get; set; } = "";
        public string Sets { get; set; } = "";
        public string PicturePath { get; set; } = "";
        public string VideoPath { get; set; } = "";
        
        public virtual ICollection<Workout> Workouts { get; set; } = new List<Workout>();
    }
}

File: FATT_frontend/FATT_NEXT/fatt/interfaces/Exercise.ts
export type Exercise = {
  Id: number;
  Name: string;
  Category: string;
  Intensity: string;
  Equipment: string;
  Repetitions: string;
  Sets: string;
  PicturePath: string;
  VideoPath: string;
};

export type ExerciseIds = {
  ExerciseIds: number[];
};

export type Exercises = {
  Data: Exercise[];
};

export type ExerciseNoIdDto = {
  Name: string;
  Category: string;
  Intensity: string;
  Equipment: string;
  Repetitions: string;
  Sets: string;
  PicturePath: string;
  VideoPath: string;
};

export type ExerciseFullDto = {
  Id: number;
  Name: string;
  Category: string;
  Equipment: string;
  Intensity: string;
  Repetitions: number;
  Sets: number;
  PicturePath: string;
  VideoPath: string;
};

export type ExerciseSimpleDto = {
  Id: number;
  Name: string;
};

export type ExerciseThumbnailDto = {
  Id: number;
  Category: string;
  PicturePath: string;
};

export type ExerciseUpdateDto = {
  Intensity: string;
  Reptitions: string;
  Sets: number;
};
